File: server/src/main/java/gg/mineral/server/network/packet/play/clientbound/PlayerPositionAndLookPacket.java
package gg.mineral.server.network.packet.play.clientbound;

import gg.mineral.api.network.packet.Packet;
import io.netty.buffer.ByteBuf;

public final record PlayerPositionAndLookPacket(double x, double headY, double z, float yaw, float pitch,
        boolean onGround)
        implements Packet.OUTGOING {
    @Override
    public void serialize(ByteBuf os) {
        os.writeDouble(x);
        os.writeDouble(headY);
        os.writeDouble(z);
        os.writeFloat(yaw);
        os.writeFloat(pitch);
        os.writeBoolean(onGround);
    }

    @Override
    public byte getId() {
        return 0x08;
    }
}

File: server/src/main/kotlin/gg/mineral/server/network/packet/play/clientbound/PlayerPositionAndLookPacket.kt
package gg.mineral.server.network.packet.play.clientbound

import gg.mineral.api.network.packet.Packet
import io.netty.buffer.ByteBuf

@JvmRecord
data class PlayerPositionAndLookPacket(
    val x: Double, val headY: Double, val z: Double, val yaw: Float, val pitch: Float,
    val onGround: Boolean
) : Packet.Outgoing {
    override fun serialize(os: ByteBuf) {
        os.writeDouble(x)
        os.writeDouble(headY)
        os.writeDouble(z)
        os.writeFloat(yaw, pitch)
        os.writeBoolean(onGround)
    }

    override val id: Byte
        get() = 0x08
}
